packages/core/src/contracts/parameterizer.ts

```diff
@@ -39,34 +40,240 @@ export class Parameterizer extends BaseWrapper<ParameterizerContract> {
+      .map((e) => {
+        return {
+          propID: e.args.propID,
+          paramName: e.args.name,
+          proposedValue: e.args.value,
+        };
+      });
   }
 
   /**
-   * An unending stream of all active Reparametization challenges
-   * @param fromBlock Starting block in history for events concerning challenges
-   *                  Set to "latest" for only new events
-   * @returns currently active challenges
+   * An unending stream of the propIDs of all Reparametization proposals currently in
+   * Challenge Reveal Phase
+   * @param fromBlock Starting block in history for events. Set to "latest" for only new events
+   * @returns currently active proposals in Challenge Reveal Phase propIDs
    */
-  public reparameterizationChallenges(fromBlock: number|"latest" = 0): Observable<EthAddress> {
+  public propIDsInChallengeRevealPhase(fromBlock: number|"latest" = 0): Observable<EthAddress> {
     return this.instance
       .NewChallengeStream({}, { fromBlock })
       .map((e) => e.args.propID)
-      .concatFilter(async (propID) => this.instance.propExists.callAsync(propID));
+      .concatFilter(async (propID) => this.isPropInChallengeRevealPhase(propID));
+  }
+
+  /**
+   * An unending stream of ParamProps of all active Reparametizations Proposals in
+   * Challenge Reveal Phase
+   * @param fromBlock Starting block in history for events. Set to "latest" for only new events.
+   * @returns ParamProps for proposals currently in Challenge Reveal Phase
+   */
+  public paramPropsInChallengeRevealPhase(fromBlock: number|"latest" = 0): Observable<ParamProp> {
+    return this.instance
+      .NewChallengeStream({}, { fromBlock })
+      .concatFilter(async (e) => this.isPropInChallengeRevealPhase(e.args.propID))
+      .map((e) => {
+        return {
+          propID: e.args.propID,
+          paramName: e.args.name,
+          proposedValue: e.args.value,
+        };
+      });
+  }
+
+  /**
+   * An unending stream of the propIDs of all Reparametization proposals that can be
+   * processed right now. Includes unchallenged applications that have passed their application
+   * expiry time, and proposals with challenges that can be resolved.
+   * @param fromBlock Starting block in history for events. Set to "latest" for only new events
+   * @returns propIDs for proposals that can be updated
+   */
+  public propIDsToProcess(fromBlock: number|"latest" = 0): Observable<EthAddress> {
+    const applicationsToProcess = this.instance
+      .ReparameterizationProposalStream({}, { fromBlock })
+      .map((e) => e.args.propID)
+      .concatFilter(async (propID) => this.isPropInUnchallengedApplicationUpdatePhase(propID));
+
+    const challengesToResolve = this.instance
+      .NewChallengeStream({}, { fromBlock })
+      .map((e) => e.args.propID)
+      .concatFilter(async (propID) => this.isPropInChallengeResolvePhase(propID));
+
+    return Observable.concat(applicationsToProcess, challengesToResolve).distinct();
```

Comment:
Use `Observable.merge` here, otherwise you will only get the first stream due to it being unending and concat works in-order on streams